File: frontend/src/entity/billing/model/SubscriptionEventType.ts
export enum SubscriptionEventType {
  Created = 'subscription.created',
  Upgraded = 'subscription.upgraded',
  Downgraded = 'subscription.downgraded',
  NewBillingCycleStarted = 'subscription.new_billing_cycle_started',
  Canceled = 'subscription.canceled',
  Reactivated = 'subscription.reactivated',
  Expired = 'subscription.expired',
  PastDue = 'subscription.past_due',
  RecoveredFromPastDue = 'subscription.recovered_from_past_due',
  Refund = 'payment.refund',
  Dispute = 'payment.dispute',
}

File: backend/internal/features/billing/models/subscription_event_type.go
package billing_models

type SubscriptionEventType string

const (
	EventCreated                SubscriptionEventType = "subscription.created"
	EventUpgraded               SubscriptionEventType = "subscription.upgraded"
	EventDowngraded             SubscriptionEventType = "subscription.downgraded"
	EventNewBillingCycleStarted SubscriptionEventType = "subscription.new_billing_cycle_started"
	EventCanceled               SubscriptionEventType = "subscription.canceled"
	EventReactivated            SubscriptionEventType = "subscription.reactivated"
	EventExpired                SubscriptionEventType = "subscription.expired"
	EventPastDue                SubscriptionEventType = "subscription.past_due"
	EventRecoveredFromPastDue   SubscriptionEventType = "subscription.recovered_from_past_due"
	EventRefund                 SubscriptionEventType = "payment.refund"
	EventDispute                SubscriptionEventType = "payment.dispute"
)
